Docs/developer_guide/parameter_nodes.md

```diff
@@ -203,7 +203,106 @@ param.value = "d" # ValueError raised
 
 ### Custom Parameter Types
 
-Custom parameters can be supported with a little bit of work.
+#### parameterPack
+
+It is often useful to group related information together in structures with useful names. Another decorator, `@parameterPack` was added to make this easier. This will make the class behave in a similar manner to Python's `@dataclasses.dataclass`.
```

Comment:
May also be helpful to add a note describing the impact on the actual names of the attributes set in the the `vtkMRMLScriptedModuleNode`

Reply:
Added comment. Updated the documented code block to further explain. Added test to enforce naming.

Reply:
Also updated to separate by a dot instead of an underscore, which is probably more idiomatic.